sources/packages/api/src/services/student/student.service.ts

```diff
@@ -174,44 +186,102 @@ export class StudentService extends RecordDataModelService<Student> {
       .where("user.id= :userId ", { userId })
-      .select("student.validSIN")
+      .select("sinValidation.isValidSIN")
       .getOne();
+    return student?.sinValidation.isValidSIN;
   }
 
   /**
    * Update the SIN validation information on a student that
    * is marked with a pending validation.
-   * If the SIN validation status is already set, it needs to be
-   * set as NULL before it is processed again.
-   * @param sin
-   * @param validSIN
-   * @returns pending sin validation
+   * @param craRecord
+   * @param fileReceived
    */
   async updatePendingSinValidation(
-    sin: string,
-    validSIN: boolean,
+    verificationId: number,
+    craRecord: CRAResponseStatusRecord,
+    fileReceived: string,
   ): Promise<void> {
-    // Only allow updates on a student that has pending SIN validation.
-    const studentToUpdate = await this.repo.findOne({
-      sin,
-      validSIN: null,
-    });
-    if (studentToUpdate) {
-      studentToUpdate.validSIN = validSIN;
-      this.repo.save(studentToUpdate);
+    const student = await this.repo
+      .createQueryBuilder("student")
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .where("student.sin  = :sin", { sin: craRecord.sin })
+      .andWhere("sinValidation.id = :verificationId", { verificationId })
+      .andWhere("sinValidation.dateReceived is null")
+      .andWhere("sinValidation.fileReceived is null")
+      .andWhere("sinValidation.isValidSIN is null")
+      .select(["student", "user.id"])
+      .getOne();
+    if (student) {
+      const isValidSIN =
+        craRecord.requestStatusCode === RequestStatusCodes.successfulRequest &&
+        craRecord.matchStatusCode === MatchStatusCodes.successfulMatch;
+      return this.connection.transaction(async (transactionalEntityManager) => {
+        const updatedResult = await transactionalEntityManager
+          .getRepository(SINValidation)
+          .createQueryBuilder("sinValidations")
+          .innerJoin("sinValidations.user", "user")
+          .update<SINValidation>(SINValidation)
+          .set({
+            dateReceived: new Date(),
+            fileReceived,
+            isValidSIN,
+            requestStatusCode: craRecord.requestStatusCode,
+            matchStatusCode: craRecord.matchStatusCode,
+            sinMatchStatusCode: craRecord.sinMatchStatusCode,
+            surnameMatchStatusCode: craRecord.surnameMatchStatusCode,
+            givenNameMatchStatusCode: craRecord.givenNameMatchStatusCode,
+            birthDateMatchStatusCode: craRecord.birthDateMatchStatusCode,
+          })
+          .where("user.id = :userId", { userId: student.user.id })
+          .andWhere("dateReceived is null")
+          .andWhere("fileReceived is null")
+          .andWhere("isValidSIN is null")
+          .returning(["id"])
+          .execute();
+        if (updatedResult.raw[0].id) {
+          await transactionalEntityManager
```

Comment:
This should be removed. The student already has a relationship with the sin_validations. Once the sin_validations is updated we do not need to do anything else with the students.